File: fodmaplog.client/src/app/domain/Nutriments.ts
export class Nutriments {
    energyKcal100g: number;
    carbohydratesServing: number;
    carbohydratesUnit: number;
    carbohydrates100g: number;
    fatServing: number;
    fatUnit: number;
    fat100g: number;
    proteinServing: number;
    proteinUnit: number;
    protein100g: number;
    servingQuantity: number;
    servingQuantityUnit: string;
    energyKcalServing: number;
    fiber100g: number;
    fiberServing: number;
    fiberUnit: string;

    constructor(
        carbohydratesServing: number,
        proteinServing: number,
        fatServing: number,
        carbohydrates100g: number,
        fat100g: number,
        protein100g: number,
        carbohydratesUnit: number,
        fatUnit: number,
        proteinUnit: number,
        energyKcal100g: number,
        servingQuantity: number,
        servingQuantityUnit: string,
        energyKcalServing: number,
        fiber100g: number,
        fiberServing: number,
        fiberUnit: string

    ) {
        this.carbohydratesServing = carbohydratesServing;
        this.proteinServing = proteinServing;
        this.fatServing = fatServing;
        this.carbohydrates100g = carbohydrates100g;
        this.fat100g = fat100g;
        this.protein100g = protein100g;
        this.carbohydratesUnit = carbohydratesUnit;
        this.fatUnit = fatUnit;
        this.proteinUnit = proteinUnit;
        this.energyKcal100g = energyKcal100g;
        this.servingQuantity = servingQuantity;
        this.servingQuantityUnit = servingQuantityUnit;
        this.energyKcalServing = energyKcalServing;
        this.fiber100g = this.fiber100g
        this.fiberServing = this.fiberServing
        this.fiberUnit = this.fiberUnit
    }
}

File: DataAccess/Entities/Nutriments.cs
using Newtonsoft.Json;

namespace DataAccess.Entities
{
    public class Nutriments : BaseEntity
    {
        public float? FatServing { get; set; }

        public float? Fat100g { get; set; }

        public string? FatUnit { get; set; }

        public float? SugarsServing { get; set; }

        public float? Sugars100g { get; set; }

        public string? SugarsUnit { get; set; }

        public float? CarbohydratesServing { get; set; }

        public float? Carbohydrates100g { get; set; }
        public string? CarbohydratesUnit { get; set; }
        public float? EnergyKcal100g { get; set; }

        public float? EnergyKcalServing { get; set; }
        public string? EnergyKcalUnit { get; set; }
        public float? ProteinsServing { get; set; }
        public float? Proteins100g { get; set; }
        public string? ProteinsUnit { get; set; }        
        public float? Fiber100g { get; set; }
        public float? FiberServing { get; set; }
        public string? FiberUnit { get; set; }

    }
}